src/NServiceBus.Core/Envelopes/EnvelopeConfigExtensions.cs

```diff
@@ -0,0 +1,22 @@
+#nullable enable
+
+namespace NServiceBus;
+
+using System;
+using Features;
+
+/// <summary>
+/// Provides extensions to register envelope handlers.
+/// </summary>
+public static class EnvelopeConfigExtensions
+{
+    /// <summary>
+    /// Adds the envelope handler type.
+    /// </summary>
+    public static void AddEnvelopeHandler<THandler>(this FeatureConfigurationContext context) where THandler : class, IEnvelopeHandler
```

Comment:
I looked into adding (or changing this) an `AddEnvelopeHandler` overload that accepts an instance to avoid having the envelope handlers dependencies go throguh DI. However the problem is that those handlers for CloudEvents depend on some metrics-related thing that depends on the `IMeterFactory` that is in DI.

https://github.com/Particular/NServiceBus.Envelope.CloudEvents/pull/22/files#diff-3c286ae3983d043ead4d23f8fb296af6cb4eb99a0cfdd0fe56464ce6e401ff93R16